### PR TITLE
docs: Add RFC for secondary indexes

### PR DESCRIPTION
**Pull request by garrensmith**
## Describe your changes

This is the RFC for secondary indexes in Tigris

### CODE REVIEW

File: rfcs/001-secondary-indexes.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
+## Introduction
```
Comment (efirs) [inline rfcs/001-secondary-indexes.md:1]: Maybe move it inside docs directory?
`docs/rfcs/...`

Comment (himank) [reply rfcs/001-secondary-indexes.md:1]: +1 I think moving it inside docs is better.

**Comment by ovaistariq** [comment]
This is great. Like @himank and @efirs commented, let's move the `rfcs` directory under `docs` directory.

File: rfcs/001-secondary-indexes.md (hunk 1 above)
Comment (garrensmith) [reply rfcs/001-secondary-indexes.md]: Good idea. I've moved it under the docs.

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**